tests/Unit/Adapter/Configure/AdvancedParameters/Administration/GeneralConfigurationTest.php

```diff
@@ -0,0 +1,164 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace Tests\Unit\Adapter\Configure\AdvancedParameters\Administration;
+
+use Cookie;
+use PHPUnit\Framework\MockObject\MockObject;
+use PrestaShop\PrestaShop\Adapter\GeneralConfiguration;
+use PrestaShop\PrestaShop\Core\Domain\Shop\ValueObject\ShopConstraint;
+use PrestaShopBundle\Form\Admin\Configure\AdvancedParameters\Administration\GeneralType;
+use Symfony\Component\OptionsResolver\Exception\InvalidOptionsException;
+use Symfony\Component\OptionsResolver\Exception\UndefinedOptionsException;
+use Tests\TestCase\AbstractConfigurationTestCase;
+
+class GeneralConfigurationTest extends AbstractConfigurationTestCase
+{
+    /**
+     * @var Cookie|MockObject
+     */
+    private $mockCookie;
+
+    /**
+     * @var GeneralConfiguration
+     */
+    private $generalConfiguration;
+
+    private const VALID_CONFIGURATION = [
+        GeneralType::FIELD_CHECK_MODULES_UPDATE => true,
+        GeneralType::FIELD_CHECK_IP_ADDRESS => true,
+        GeneralType::FIELD_FRONT_COOKIE_LIFETIME => Cookie::FRONT_LIFETIME,
+        GeneralType::FIELD_BACK_COOKIE_LIFETIME => Cookie::BACK_LIFETIME,
+        GeneralType::FIELD_COOKIE_SAMESITE => Cookie::SAMESITE_LAX,
+    ];
+
+    private const SHOP_ID = 42;
+
+    protected function setUp(): void
+    {
+        parent::setUp();
+
+        $this->mockCookie = $this->createCookieMock();
+
+        $this->generalConfiguration = new GeneralConfiguration(
+            $this->mockConfiguration,
+            $this->mockShopConfiguration,
+            $this->mockMultistoreFeature,
+            $this->mockCookie
+        );
+    }
+
+    /**
+     * @return MockObject|Cookie
+     */
+    protected function createCookieMock()
```

Comment:
```suggestion
    protected function createCookieMock(): Cookie
```

Reply:
Return type is MockObject|Cookie, so it's mixed.